gcapi/gcapi.py

```diff
@@ -243,15 +347,32 @@ def __call__(self,
             extra_headers["Content-Type"] = "application/json"
 
         self._validate_url(url)
+        if method == "GET":
```

Comment:
This should not be done with an if statement, you should be able to achieve this with variable unpacking.